### PR TITLE
varnish4: init at 4.1.9; varnish6: init at 6.0.0

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

added ```Varnish 6``` (new version) and ```Varnish 4``` (legacy version still supported by upstream) in additional to ```Varnish 5```.

### CODE REVIEW

**Comment by volth** [comment]
@GrahamcOfBorg build varnish4

**Comment by volth** [comment]
@GrahamcOfBorg build varnish6

**Comment by Mic92** [comment]
```
@GrahamcOfBorg build collectd varnish4 collectd-data varnish varnish6 varnish5
```

**Comment by Mic92** [comment]
```
@GrahamcOfBorg build varnish4Packages.digest varnish4Packages.rtstatus varnish4Packages.modules varnish4Packages.geoip varnish5Packages.digest varnish5Packages.dynamic varnish5Packages.modules varnish5Packages.geoip varnish6Packages.digest
```

**Comment by volth** [comment]
Should this be patched for Darwin?
https://github.com/nigoroll/libvmod-dynamic/blob/23580af18838327c6ae4b9d5fdff2494a81398a4/autogen.sh#L8-L9

**Comment by volth** [comment]
collectd failure seems inrrelevant:
```
./lib/riemann/attribute.h:21:10: fatal error: riemann/proto/riemann.pb-c.h: No such file or directory
 #include <riemann/proto/riemann.pb-c.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by Mic92** [comment]
I just fixed riemann:

```
@GrahamcOfBorg build collectd 
```

File: pkgs/servers/varnish/dynamic.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    repo = "libvmod-dynamic";
+    rev = "475be183fddbd727c3d2523f0518effa9aa881f8"; # 5.2 branch for Varnish-5.2 https://github.com/nigoroll/libvmod-dynamic/commits/5.2
+    sha256 = "12a42lbv0vf6fn3qnvngw893kmbd006f8pgab4ir7irc8855xjgf";
+  };
+
+  nativeBuildInputs = [ pkgconfig docutils automake autoconf libtool ];
+  buildInputs = [ varnish python ];
+  preConfigure = ''
+    substituteInPlace autogen.sh --replace 'pkg-config --variable=datarootdir varnishapi' 'echo ${varnish.dev}/share'
+    ./autogen.sh
```
Comment (Mic92) [inline pkgs/servers/varnish/dynamic.nix]: do we actually need to run their `autogen.sh`? It seems not to do much more then `autoreconfHook`.

Comment (volth) [reply pkgs/servers/varnish/dynamic.nix]: yes, changed to ```autoreconfHook```

**Comment by Mic92** [comment]
```
@GrahamcOfBorg build varnish4Packages.digest varnish4Packages.rtstatus varnish4Packages.modules varnish4Packages.geoip varnish5Packages.digest varnish5Packages.dynamic varnish5Packages.modules varnish5Packages.geoip varnish6Packages.digest
```

**Pull request merged**